agent/agent.go

```diff
@@ -228,21 +228,9 @@ func New(c *Config) (*Agent, error) {
 		httpAddrs:       httpAddrs,
 		tokens:          new(token.Store),
 	}
-	if err := a.resolveTmplAddrs(); err != nil {
```

Comment:
The tagged addresses won't be set until line 275, so I think the treatment of them won't work (and never did). It seems like we should get rid of the template processing for those, and maybe move everything in here up to line 258 over in the config where you moved the template processing (right after processing).